src/code/containers/challenge-container-selector.js

```diff
@@ -43,27 +44,33 @@ class ChallengeContainerSelector extends Component {
   }
 
   componentWillMount() {
-    const { routeParams, navigateToCurrentRoute, navigateToChallenge } = this.props;
-    if (hasChangedRouteParams(this.props)) {
+    const { navigateToCurrentRoute, navigateToChallenge, authoring } = this.props;
+    // the URL's challengeId is only used for initial routing, so prioritize the numeric route params
```

Comment:
Looking at this, another enhancement that comes to mind would be allowing partial route specification, e.g. `#/level/mission` would go to the first challenge of the specified mission. The relevance to this code is that to support `#/level` as a shortcut for the first mission of the specified level, then the corresponding line in `app.js` would be something like:
```
<Route path="/(:challengeIdOrLevel)" component={ChallengeContainerSelector} />
```
and then the `ChallengeContainerSelector` could test the type of the value to determine whether it's a challengeId (string) or level (number). That's out of scope for this PR, however, so I'll write it up as a PT story.